### ISSUE TITLE
Docker image won't start https without --address 

### ISSUE THREAD

**Issue by eco-minio** [minio/minio#7887]
When starting a container via https, certs are not read properly if we do not specify --address flag
## Expected Behavior

MinIO container should be able to start https without mandating --address
## Current Behavior

Without --address (https doesn't work):
```
docker run -e MINIO_ACCESS_KEY=minio -e MINIO_SECRET_KEY=minio123 --rm -p 9000:9000 --name tlstest -v /tmp/data:/dhome/e/.minio:/root/.minio minio/minio:latest server /data

Endpoint:  http://172.17.0.2:9000  http://127.0.0.1:9000
```

With --address (https works):
```
docker run -e MINIO_ACCESS_KEY=minio -e MINIO_SECRET_KEY=minio123 --rm -p 9000:9000 --name tlstest -v /tmp/data:/data -v /home/e/.minio:/root/.minio minio/minio:latest server --address=:9000 /data
2019/07/05 21:05:58 Invalid total number of endpoints for erasure mode

Endpoint:  https://172.17.0.2:9000  https://127.0.0.1:9000
```

## Steps to Reproduce (for bugs)

Run docker commands from above

## Regression

Not sure

## Your Environment

* Version used (`minio version`): minio/latest
* Operating System and version (`uname -a`): Ubuntu 18.04

**Comment by harshavardhana** [comment]
This issue is related to docker container running as non root uswr., We need a comprehensive way to handle this.

**Comment by harshavardhana** [comment]
Fixed by #7930
